### PR TITLE
Generate the  build number at the beginning of the build

### PR DESCRIPTION
**Pull request by jakubgs**
Generate the build number at the beginning of the build to avoid mismatched numbers.

Resolves. https://github.com/status-im/status-react/issues/6881

### CODE REVIEW

**Review by jeluard — APPROVED** [review]

File: scripts/gen_build_no.sh [hunk 1]
```diff
@@ -20,7 +20,7 @@ BUILD_NUMBER_FILE="${GIT_ROOT}/BUILD_NUMBER"
 if [[ -f "${BUILD_NUMBER_FILE}" ]]; then
     cat "${BUILD_NUMBER_FILE}"
 else
-    # Format: Year(2 digit) + Month + Day + Hour + Minutes
-    # Example: 1812011805
-    date '+%y%m%d%H%M' | tee "${BUILD_NUMBER_FILE}"
+    # Format: Year(2 digit) + Month + Day + Hour
+    # Example: 2018120118
```
Comment (mandrigin) [inline scripts/gen_build_no.sh:24]: The comment is wrong now. Why this change, btw?

**Review by mandrigin — CHANGES_REQUESTED** [review]

File: ci/Jenkinsfile.android [hunk 2]
```diff
@@ -35,7 +35,10 @@ pipeline {
           /* Necessary to load methods */
           mobile = load 'ci/mobile.groovy'
           cmn    = load 'ci/common.groovy'
+          /* Cleanup and Prep */
           mobile.prep(cmn.getBuildType())
+          /* Run at start to void mi-matched numbers */
```
Comment (pedropombeiro) [inline ci/Jenkinsfile.android]: typo in mismatched

File: ci/Jenkinsfile.ios [hunk 3]
```diff
@@ -31,7 +31,10 @@ pipeline {
           /* Necessary to load methods */
           mobile = load 'ci/mobile.groovy'
           cmn    = load 'ci/common.groovy'
+          /* Cleanup and Prep */
           mobile.prep(cmn.getBuildType())
+          /* Run at start to void mi-matched numbers */
```
Comment (pedropombeiro) [inline ci/Jenkinsfile.ios]: Same here

**Review by pedropombeiro — CHANGES_REQUESTED** [review]

**Comment by Serhy** [comment]
@jakubgs I've checked Android and iOS builds from https://github.com/status-im/status-react/pull/6894#issuecomment-442147029 and it's good! Both apps has (2018112717) version. 

Could you please review comments above about comments (sorry for tautoligies! :) inside PR so we could proceed as we need this changes in 0.9.32 release.

File: scripts/gen_build_no.sh (hunk 1 above)
Comment (jakubgs) [reply scripts/gen_build_no.sh:24]: I've removed minutes to reduce possibility of getting different build numbers for different platforms.

File: ci/Jenkinsfile.android (hunk 2 above)
Comment (jakubgs) [reply ci/Jenkinsfile.android]: fixed

File: ci/Jenkinsfile.ios (hunk 3 above)
Comment (jakubgs) [reply ci/Jenkinsfile.ios]: fixed

File: scripts/gen_build_no.sh (hunk 1 above)
Comment (jakubgs) [reply scripts/gen_build_no.sh:24]: Also, fixed comment.

**Review by mandrigin — APPROVED** [review]
lgtm now

**Review by pedropombeiro — APPROVED** [review]
